drafttopic/utilities/tests/test_fetch_wikiprojects.py

```diff
@@ -0,0 +1,113 @@
+from nose.tools import eq_
+from ..fetch_wikiprojects import WikiProjectsParser
+from ..fetch_wikiprojects import WPDPage
+import logging
+import json
+import os
+import pdb
+
+logger = logging.getLogger( __name__ )
+logging.basicConfig(level=logging.INFO)
+
+excluded_keys = ['topics', 'root_url', 'index', 'name', 'url']
+
+def test_WikiProjectsFromTable():
+    table = fetchTestSectionText('performing_arts.test')
+    parser = WikiProjectsParser(WPDPage)
+    topics = parser.getWikiProjectsFromTable(table)
+    topics = {'topics': topics}
+    actualTopics = {'topics':
+                    {
+                        'Circus': {'name': 'Wikipedia:WikiProject Circus'},
+                        'Magic (close-up magic)':
+                            {'name': 'Wikipedia:WikiProject Magic'},
+                        'Magic (stage magic)': {'name': 'Wikipedia:WikiProject Magic'},
+                        'Reenactment': {'name': 'Wikipedia:WikiProject Reenactment'}
+                    }
+                   }
+    wpTopicsCompare(actualTopics, topics)
+
+def test_getWikiProjectsFromSectionIntroText():
+    parser = WikiProjectsParser(WPDPage)
```

Comment:
This parser is actually making an HTTP connection.  That means the test could fail if you don't have a connection to the internet.  It's better to download the page content and include the data in the test so that an internet connection is not necessary.

Reply:
Only the test - test_get_sub_categories makes api calls while all others use offline data.
Even the above test has been made to cache data offline in case tests are repeated. However, I'll still think about how can we mock api calls, as the method tested by this test makes repeated api calls for sections rather than fetching a page in a single call.